### PR TITLE
Recv maxattempts

### PR DESCRIPTION
**Pull request by robertfarnum**
Attempts implementation (Issue #60):
1) Updated spec.go to test whether the expected number of attempts has been reached
2) Updated manual.md to document attempts
3) Added new attempts.yaml demo

### CODE REVIEW

File: demos/shell.yaml [hunk 1]
```diff
@@ -64,12 +64,13 @@ spec:
...
             payload: |
-              exit 1
+              ls ?
+              exit $?
         - recv:
             doc: |
               Handle the shell exit code 1
             topic: exit
-            regexp: 1
+            regexp: "2"
```
Comment (carlos-loya) [inline demos/shell.yaml:73]: I get a failure on this line when `regexp` is set to `2`

**Review by carlos-loya — APPROVED** [review]
Built and tested locally. I ran into an issue when running `plax -test demos/shell.yaml` where it would fail the last `recv` on `phase1`. It works when I changed it back to `1`. Apart from that, looks great!

File: demos/shell.yaml (hunk 1 above)
Comment (robertfarnum) [reply demos/shell.yaml:73]: This should be fixed

**Review by carlos-loya — APPROVED** [review]
Thanks for adding this feature. I played around with the `attempts.yaml` and it works as expected!

**Pull request merged**